extensions/amp-list/0.1/amp-list.js

```diff
@@ -340,40 +345,85 @@ export class AmpList extends AMP.BaseElement {
     );
   }
 
+  /**
+   * Returns true if element's src points to amp-state.
+   *
+   * @return {boolean}
+   * @private
+   */
+  isAmpStateSrc_() {
+    const src = this.element.getAttribute('src');
+    return (
+      isExperimentOn(this.win, 'amp-list-init-from-state') &&
+      typeof src === 'string' &&
+      startsWith(src, 'amp-state:')
+    );
+  }
+
+  /**
+   * Returns json pointed at by an amp-state protocol src. For example,
+   * src="amp-state:json.path".
+   *
+   * @return {Promise<!Object|undefined>}
+   * @private
+   */
+  getAmpStateJson_() {
+    const src = this.element.getAttribute('src');
+    const ampStatePath = src.substring('amp-state:'.length);
```

Comment:
Nit: Use the raw integer length and add a comment. Or share the `'amp-state:'` string by hoisting it.

Reply:
Done.